AtSheet/explainer.md

```diff
@@ -0,0 +1,225 @@
+```
+
+#### Performance
+
+This will be a performance-neutral feature, and use of it may allow for developers to reduce the number of network requests. We should ensure that multiple imports of different sheets from the same file produce a single network request. 
+
+```js
+// The following two imports should only make a single network request.
+import { foo } from 'sheet.css' with { type: 'css' };
+import { bar } from 'sheet.css' with { type: 'css' }
+```
+
+```html
+<style>
+/* The following two imports should only make a single network request. */
+@import "sheet.css#foo";
+@import "sheet.css#bar";
+</style>
+```
+
+```html
+<!-- The following two link tags should only make a single network request. -->
+<link rel="stylesheet" href="sheet.css#foo" />
+<link rel="stylesheet" href="sheet.css#bar" />
+```
+
+#### Interaction with CSSOM
+
+
+Named `@sheet` references augment the [existing](https://drafts.csswg.org/cssom/#stylesheet) `StyleSheet` interface with an optional `name` attribute reflecting the `@sheet` identifier:
+
+```
+[Exposed=Window]
+interface StyleSheet {
+  readonly attribute DOMString? name;
+};
+```
+*Open issue: Should this overload the existing `title` attribute instead?*
+
+This also expands the [existing](https://drafts.csswg.org/cssom/#cssstylesheet) CSSOM `CSSStyleSheet` definition with a `StyleSheetList` of nested `CSSStyleSheet` objects to access nested `@sheet` references:
+
+```
+[Exposed=Window]
+interface CSSStyleSheet : StyleSheet {
+  [SameObject] readonly attribute StyleSheetList nestedStyleSheets;
+};
+```
+*Open issue: The name `nestedStyleSheets` is up for discussion*
+
+## Considered alternatives
+
+1. [Declarative CSS Modules](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/ShadowDOM/explainer.md) are another mechanism for sharing styles between Declarative Shadow DOM and light DOM without the use of Javascript.
+
+## Open Issues
+
+1. Whether rules are applied automatically for `@sheet` definitions, or whether they need to be imported to apply. The CSS Working Group did not have a consensus.
+2. Fragment-only identifiers (without a URL) should allow inline `@sheet` references on the same document to be included globally (even within shadow roots). This wasn't brought up in the CSSWG discussions at all, but is important for DSD without requiring an external file (to avoid FOUC).
+3. Behavior of `@import` - should this be possible within `@sheet` at all, should it be allowed if it's the first/only statement, or should it be blocked? There was discussion of this in the CSSWG, but no conclusion was reached.
+4. What happens with multiple `@sheet` definitions with the same identifier? First-definition wins, or do they get merged like `@layer`? Again, this was brought up in the CSSWG but not resolved. Note that it's possible to have a "Flash of other-styled content" if it's last-defintion-wins, as the first definition may apply, then a later definition may override it.
```

Comment:
Why would this lead to flash of other styled content? Since we apply the full cascade before painting, could that actually happen?

Reply:
i.e. I would presume if a DSD uses the same name for the sheet, we would want the last to win instead of the first (unless they combine)

Reply:
> Why would this lead to flash of other styled content? Since we apply the full cascade before painting, could that actually happen?

Anything in a separate CSS file can potentially cause an FOUC if network latency is high enough (and it's not in the cache).

Reply:
For #4 specifically, if the final duplicated `@sheet` definition is in a separate CSS file, the document could paint before download has completed, then need to repaint with the overridden definition.

Reply:
Ah gotcha, I was thinking if they were in the same file